src/agent/src/agent.cpp

```diff
@@ -42,6 +42,18 @@ Agent::Agent(const std::string& configFilePath, std::unique_ptr<ISignalHandler>
 
     m_taskManager.Start(
         m_configurationParser.GetConfig<size_t>("agent", "thread_count").value_or(config::DEFAULT_THREAD_COUNT));
+
+    m_commandHandler.CleanUpInProgressCommands(
```

Comment:
I think I prefer to do this at the beginning of `CommandsProcessingTask`. If so, function can be private.

Reply:
Done!